### PR TITLE
Port to Qt5

### PR DESCRIPTION
**Pull request by dschmidt**
Hi,

this allows building QtSparkle with Qt4 and Qt5.
I bumped the CMake version requirement to keep the changes minimal (Qt5 target support, Automoc, Autouic).

Regards,
Domme

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
+cmake_policy(SET CMP0043 NEW)
 
 include(cmake/Translations.cmake)
 
-find_package(Qt4 REQUIRED QtCore QtGui QtNetwork QtXml)
-add_definitions(${QT_DEFINITIONS})
-include(${QT_USE_FILE})
+option(BUILD_WITH_QT4 "Build qtsparkle with Qt4" ON)
+if(BUILD_WITH_QT4)
+    find_package(Qt4 REQUIRED QtCore QtGui QtNetwork QtXml)
```
Comment (davidsansome) [inline CMakeLists.txt]: Can you use 2-space indentation here to match the convention elsewhere?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -86,11 +69,16 @@ compile_translations(SOURCES
 
 add_library(qtsparkle
   ${SOURCES}
-  ${MOC}
-  ${UIC}
   ${TRANSLATION_TEMPLATE}
 )
 
+set_target_properties(qtsparkle
+    PROPERTIES
```
Comment (davidsansome) [inline src/CMakeLists.txt]: Here too

**Comment by davidsansome** [comment]
Thanks so much for doing this!

File: CMakeLists.txt (hunk 1 above)
Comment (dschmidt) [inline CMakeLists.txt]: Sure, sorry, was already late :)

**Comment by dschmidt** [comment]
Fixed intendation, made Qt4 and Qt5 version completely coinstallable by also suffixing the include dir and removed some left over references to ${MOC_SOURCES} ${UI_SOURCES}.

**Pull request merged**